lib.py

```diff
@@ -1,167 +1,284 @@
+        logger.debug("Build container output: %s", out)
         return True
-    except docker.errors.BuildError as err:
-        logger.exception("Build Error while building for %s", msg.author.name)
-        e = ""
-        for chunk in err.build_log:
-            e += chunk.get("stream") or ""
-        await msg.reply(f"Error building benchmark: {err}")
+    except docker.errors.ContainerError:
+        logger.exception("Error in docker while building code")
         return False
-    finally:
-        await status.delete()
-
 
-async def run_image(msg, aoc_input):
-    logger.info("Running image/benchmark for %s", msg.author.name)
-    # input = ','.join([str(int(x)) for x in input])
-    status = await msg.reply("Running benchmark...")
-    loop = asyncio.get_event_loop()
-    try:
-        out = await loop.run_in_executor(
-            None,
-            functools.partial(
-                doc.containers.run,
-                f"ferris-elf-{msg.author.id}",
-                f"timeout 60 ./target/release/ferris-elf",
-                environment=dict(INPUT=aoc_input),
-                remove=True,
-                stdout=True,
-                mem_limit="24g",
-                network_mode="none",
-            ),
+def load_answers(cursor: Cursor, day: int, part: int):
+    """Load the expected answers for each input file."""
+    rows = cursor.execute(
+            "SELECT key, answer2 FROM solutions WHERE day = ? AND part = ?",
+            (day, part),
         )
-        out = out.decode("utf-8")
-        print(out)
-        return out
-    except docker.errors.ContainerError as err:
-        logger.exception("Error while running image for %s", msg.author.name)
-        await msg.reply(f"Error running benchmark: {err}")
-    finally:
-        await status.delete()
 
+    answer_map = {}
+    for r in rows:
+        (key, answer) = r
+        answer_map[key] = answer
 
-async def benchmark(submit_msg: SubmitMessage):
-    (msg, day, code, part) = submit_msg.msg, submit_msg.day, submit_msg.code, submit_msg.part
-    build = await build_image(msg, code)
-    if not build:
-        return
+    return answer_map
+
+def get_input_files(day: int):
+    """
+    List all the input files that exist for the current day.
+    """
+    day_path = get_input_dir_for_day(day)
+    return [f for f in os.listdir(day_path) if os.path.isfile(os.path.join(day_path, f))]
+
+def load_input(tmp_dir: str, day: int, file_name: str):
+    """
+    Populate tmp_dir with the input files for the requested year/day.
+    """
+    day_path = get_input_dir_for_day(day)
+    container_inputs_path = os.path.join(tmp_dir, "inputs")
+
+    # Step 1: Clear out anything there
 
-    day_path = f"{day}/"
+    for path in pathlib.Path(container_inputs_path).glob("**/*"):
+        if path.is_file():
+            path.unlink()
+        elif path.is_dir():
+            shutil.rmtree(path)
+
+    for path in pathlib.Path(container_inputs_path).glob("**/.*"):
```

Comment:
the deletion code is repeated twice here?

Reply:
Yes. It's kinda silly, but there's some confusion whether, for example, the `.keep` file would be removed from just the first portion. This is a hack, but either I did my testing wrong (probable), or it does work.